lib/CodeGen/CodeGenFunction.h

```diff
@@ -3317,6 +3317,13 @@ class CodeGenFunction : public CodeGenTypeCache {
 
   template<typename T>
   static bool isObjCMethodWithTypeParams(const T *) { return false; }
+
+  // Determine whether the given argument is FunctionProtoType
+  // that have parameter bounds information within it
+  static bool isFunctionProtoTypeWithTypeParams(const FunctionProtoType* FPT) { return true; }
```

Comment:
I think this should be renamed to isFunctionProtoType - there's really nothing about type parameters here.